Access Engineering.js

```diff
@@ -0,0 +1,334 @@
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	let title = attr(doc, 'meta[name="citation_title"]', 'content');
+	if (title) {
+		if (doc.querySelector('meta[name="citation_isbn"]')) {
+			let bookTitle = attr(doc, 'meta[name="citation_book_title"]', 'content');
+			if (!bookTitle || title == bookTitle) {
+				return "book";
+			}
+			else {
+				return "bookSection";
+			}
+		} else if(url.includes('content/video/')) {
+			return 'videoRecording'; 
+		} else {
+			return "journalArticle";
+		}
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.results-item a[href]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url);
+	}
+}
+
+function scrape(doc, url) {
+	// EM is a provides better rule. Most of the empty field 
+	// filled from EM and its missing editions 
+	// for books and books chapter page - so we'll fill
+	// those in manually.
+	var translator = Zotero.loadTranslator('web');
+	// Embedded Metadata
+	translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+	translator.setDocument(doc);
+
+	translator.setHandler('itemDone', function (obj, item) {
+		// Edition
+		let edition = ZU.xpathText(doc, '//meta[@name="citation_edition"]/@content');
+		if(edition) item.edition = edition;
+
+		item.complete();		
+	});
+
+	translator.getTranslatorObject(function(trans){
+		if(detectWeb(doc, url) == "bookSection") {
+			trans.itemType = "bookSection";
+		}		
+		trans.addCustomFields({
+			citation_book_title: "bookTitle"
+		});		
+		trans.doWeb(doc, url);
+	});
+}
+/** BEGIN TEST CASES **/
+var testCases = [
+	{
+		"type": "web",
+		"url": "https://www.accessengineeringlibrary.com/content/book/9781259643835",
+		"items": [
+			{
+				"itemType": "book",
+				"title": "Automatic Control Systems",
+				"creators": [
+					{
+						"firstName": "Dr Farid",
```

Comment:
Remove prefixes

Reply:
@AbeJellinek - Could you please explain more about remove prefixes in above test data shown? What exactly need to update?

Reply:
"Dr" here, "Professor", "B. S. E. E.", and "P. E." elsewhere. Academic/professional titles don't go in item metadata.

Reply:
Fixed.